Babylon/utils/request.py

```diff
@@ -0,0 +1,47 @@
+import logging
+from typing import Any
+from typing import Optional
+
+import requests
+
+logger = logging.getLogger("Babylon")
+
+
+def oauth_request(url: str,
+                  access_token: str,
+                  data: Any = {},
+                  json_data: Any = "",
+                  params: Any = {},
+                  type: str = "GET") -> Optional[Any]:
+    """Requests an API using OAuth authentication
```

Comment:
I'm not sure that a oauth_request function is really useful, while it is true that most code doing such request is mostly the same for the call part, the treatment could change from end-point to end-point.

Reply:
For the powerbi example it saves a lot of duplicate code so I would say it's beneficial.
For particular cases, i still use raw requests.